### ISSUE TITLE
`nn.tabulate()` unexpectedly prints to the jupyter console

### ISSUE THREAD

**Issue by grahamgower** [google/flax#2359]
### System information
flax 0.5.3
rich 11.2.0

### Problem you have encountered:
The new `tabulate()` function returns a string. It doesn't print the tabulation to the terminal. However, if the function is run inside a jupyter notebook, the table is printed to the jupyter console when the string is generated. This is problematic for me, because the table then appears inside jupyter-book documentation even though I didn't request it, and there doesn't appear to be a way to supress the output.

### What you expected to happen:
The same behaviour as in a regular python terminal---nothing printed unless I call `print(s)` on the returned string `s`.

### Example

Run the following in a regular `python` terminal and nothing is printed.
Run it in a jupyter-lab ipython cell and the table appears.
```python
import jax
import jax.numpy as jnp
import flax.linen as nn

class Foo(nn.Module):
    @nn.compact
    def __call__(self, x):
        h = nn.Dense(4)(x)
        return nn.Dense(2)(h)

x = jnp.ones((16, 9))
tabulate_fn = nn.tabulate(Foo(), jax.random.PRNGKey(0))
s = tabulate_fn(x)
```
If you run that example code in a jupyter notebook, the variable `s` has the value `"\n\n"`, which doesn't quite seem right.

### Probable fix
Maybe this is fixed by adding `force_jupyter=False` to the Console constructor?
https://github.com/google/flax/blob/48dfc05730f09868d511fa907da7e081ef316ca3/flax/linen/summary.py#L412-L416

**Comment by cgarciae** [comment]
Hey @grahamgower! Thanks for reporting this. I was checking `Console` and it seems there is a `force_jupyter` flag which when set to `False` might fix this issue, will test it and send a patch if works.

**Comment by cgarciae** [comment]
This should be fixed once #2316 is merged.
